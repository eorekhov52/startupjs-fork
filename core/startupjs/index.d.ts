export { default as BASE_URL } from '@startupjs/utils/BASE_URL'
export { default as axios } from '@startupjs/utils/axios'
export * from 'teamplay'
export * from '@startupjs/hooks'
// HINT: `isomorphic` means that the code can be executed both
//        on the server and on the client
export * from '@startupjs/isomorphic-helpers'
// dummy babel macro functions for @startupjs/babel-plugin-rn-stylename-inline
export function css (css: TemplateStringsArray): any
export function styl (styl: TemplateStringsArray): any
export function pug (pug: TemplateStringsArray): any

export { default as StartupjsProvider } from './StartupjsProvider.js'
