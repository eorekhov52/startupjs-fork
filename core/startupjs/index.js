export { default as BASE_URL } from '@startupjs/utils/BASE_URL'
export { default as axios } from '@startupjs/utils/axios'
export * from 'teamplay'
export * from '@startupjs/hooks'
// HINT: `isomorphic` means that the code can be executed both
//        on the server and on the client
export * from '@startupjs/isomorphic-helpers'
// dummy babel macro functions for @startupjs/babel-plugin-rn-stylename-inline.
export function css (cssString) { return cssString }
export function styl (stylString) { return stylString }
// dummy pug function (it gets compiled to jsx by babel pug plugin)
export function pug (pugString) { return pugString }
export { default as StartupjsProvider } from './StartupjsProvider.js'

// loading config should be performed first
export { default as __dummyLoadConfig } from '@startupjs/registry/loadStartupjsConfig.auto'

// init client at the very end. This is hacky solution to handle import loops
export { default as __dummyInitClient } from './initClient.auto.js'
