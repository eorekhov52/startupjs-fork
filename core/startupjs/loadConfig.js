// manually load the startupjs config instead of using the one which is
// automatically loaded from the startupjs.config.js file in the project root.
// This is useful when using startupjs in client-only mode and without its compilation pipeline.
export { default } from '@startupjs/registry/loadStartupjsConfig'
