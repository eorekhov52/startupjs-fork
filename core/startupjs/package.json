{
  "name": "startupjs",
  "type": "module",
  "version": "0.56.0-alpha.69",
  "engines": {
    "node": ">= 14"
  },
  "description": "React Native + Web + Node.js full-stack framework",
  "main": "index.js",
  "exports": {
    ".": "./index.js",
    "./server": "./server.js",
    "./startServer": "./startServer.js",
    "./registry": "./registry.js",
    "./initClient": "./initClient.js",
    "./nconf": "./nconf.js",
    "./isomorphicHelpers": "./isomorphicHelpers.js",
    "./loadConfig": "./loadConfig.js",
    "./metro-config": "./metro-config.cjs",
    "./metro-babel-transformer": "./metro-babel-transformer.cjs",
    "./babel": "./babel.cjs",
    "./startupjs.config": "./startupjs.config.cjs",
    "./package.json": "./package.json",
    "./plugins/cssMediaUpdater.plugin": "./plugins/cssMediaUpdater.plugin.js"
  },
  "bin": "./cli.js",
  "license": "MIT",
  "dependencies": {
    "@startupjs/babel-plugin-rn-stylename-to-style": "^0.56.0-alpha.34",
    "@startupjs/bundler": "^0.56.0-alpha.69",
    "@startupjs/cli": "^0.56.0-alpha.65",
    "@startupjs/hooks": "^0.56.0-alpha.64",
    "@startupjs/isomorphic-helpers": "^0.56.0-alpha.0",
    "@startupjs/registry": "^0.56.0-alpha.47",
    "@startupjs/server": "^0.56.0-alpha.69",
    "@startupjs/utils": "^0.56.0-alpha.64",
    "babel-preset-startupjs": "^0.56.0-alpha.69",
    "lodash": "^4.17.20",
    "teamplay": "^0.1.4"
  },
  "peerDependencies": {
    "react": "*",
    "react-native": "*"
  },
  "peerDependenciesMeta": {
    "react-native": {
      "optional": true
    }
  }
}
