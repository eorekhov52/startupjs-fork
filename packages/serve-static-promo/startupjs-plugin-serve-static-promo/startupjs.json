{
  "type": "plugin",
  "name": "startupjs-plugin-serve-static-promo",
  "for": "startupjs"
}
