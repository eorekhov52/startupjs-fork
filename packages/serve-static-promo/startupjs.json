{
  "type": "module",
  "name": "serve-static-promo",
  "bundler": {
    "forceCompile": {
      "web": true,
      "server": true
    }
  },
  "imports": [
    "./startupjs-plugin-serve-static-promo"
  ]
}
