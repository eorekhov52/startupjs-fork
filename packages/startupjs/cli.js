#!/usr/bin/env node
'use strict'

import { run } from '@startupjs/cli'

run()
