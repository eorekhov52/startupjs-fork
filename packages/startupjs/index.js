// Make ShareDB client and Racer client work in React Native and in Webpack
import dummyMockBrowserify from '@startupjs/utils/mockBrowserify'
import dummyInit from '@startupjs/init/client-auto'

export { default as axios } from '@startupjs/utils/axios'
export * from '@startupjs/react-sharedb'
export { default as $, signal } from '@startupjs/signals'
export * from '@startupjs/hooks'
// HINT: `isomorphic` means that the code can be executed both
//        on the server and on the client
export * from '@startupjs/isomorphic-helpers'
// dummy babel macro functions for @startupjs/babel-plugin-rn-stylename-inline.
export function css (cssString) { return cssString }
export function styl (stylString) { return stylString }
// dummy pug function (it gets compiled to jsx by babel pug plugin)
export function pug (pugString) { return pugString }
export { default as t } from '@startupjs/i18n/client/t.js'

// this library is pure side-effects, so we force usage of a dummy function
// to prevent dead-code elimination
dummyMockBrowserify()
dummyInit() // prevent dead-code elimination
