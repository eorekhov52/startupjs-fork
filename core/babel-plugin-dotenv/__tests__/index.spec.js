const pluginTester = require('babel-plugin-tester').default
const plugin = require('../index')
const { name: pluginName } = require('../package.json')

pluginTester({
  plugin,
  pluginName,
  snapshot: true,
  pluginOptions: {
    path: ['__tests__/.env']
  },
  tests: {
    'Transforms imports to constants, and places them after all other imports': /* js */`
      import { observer } from 'startupjs'
      import { BASE_URL, STRIPE_PUBLIC_KEY } from '@env'
      import React from 'react'
      import { Text } from 'react-native'
      const foo = 'Bar' + BASE_URL
    `,
    'Transforms when @env is a sole import': /* js */`
      import { BASE_URL, STRIPE_PUBLIC_KEY } from '@env'
      const foo = 'Bar' + BASE_URL
    `,
    'Override value with mock': {
      pluginOptions: {
        path: ['__tests__/.env'],
        override: {
          BASE_URL: "typeof window !== 'undefined' && window.location && window.location.origin"
        }
      },
      code: /* js */`
        import { BASE_URL as baseUrl } from '@env'
        const foo = 'Bar' + baseUrl
      `
    },
    'When override is present, do it no matter if the env exists or not': {
      pluginOptions: {
        path: [],
        override: {
          BASE_URL: "typeof window !== 'undefined' && window.location && window.location.origin"
        }
      },
      code: /* js */`
        import { BASE_URL as baseUrl } from '@env'
        const foo = 'Bar' + baseUrl
      `
    },
    'Overrides value when multiple envs specified': {
      pluginOptions: {
        path: ['__tests__/.env', '__tests__/.env.production']
      },
      code: /* js */`
        import { BASE_URL, STRIPE_PUBLIC_KEY } from '@env'
        const foo = 'Bar' + BASE_URL
      `
    },
    'Throws an error if the var is not present': {
      code: /* js */`
        import { BASE_URL, NON_EXISTENT } from '@env'
        const foo = 'Bar' + BASE_URL
      `,
      error: true
    }
  }
})
