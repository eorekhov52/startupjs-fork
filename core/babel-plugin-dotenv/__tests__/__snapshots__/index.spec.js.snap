// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@startupjs/babel-plugin-dotenv Override value with mock: Override value with mock 1`] = `

import { BASE_URL as baseUrl } from '@env'
const foo = 'Bar' + baseUrl

      â†“ â†“ â†“ â†“ â†“ â†“

const baseUrl =
  (typeof window !== "undefined" &&
    window.location &&
    window.location.origin) ||
  "https://localhost:3000";
const foo = "Bar" + baseUrl;


`;

exports[`@startupjs/babel-plugin-dotenv Overrides value when multiple envs specified: Overrides value when multiple envs specified 1`] = `

import { BASE_URL, STRIPE_PUBLIC_KEY } from '@env'
const foo = 'Bar' + BASE_URL

      â†“ â†“ â†“ â†“ â†“ â†“

const STRIPE_PUBLIC_KEY = "000-000000-00";
const BASE_URL = "https://example.com";
const foo = "Bar" + BASE_URL;


`;

exports[`@startupjs/babel-plugin-dotenv Throws an error if the var is not present: Throws an error if the var is not present 1`] = `

import { BASE_URL, NON_EXISTENT } from '@env'
const foo = 'Bar' + BASE_URL

      â†“ â†“ â†“ â†“ â†“ â†“

SyntaxError: unknown: "NON_EXISTENT" is not defined in ["__tests__/.env"]
[0m[31m[1m>[22m[39m[90m 1 | [39m[36mimport[39m { [33mBASE_URL[39m[33m,[39m [33mNON_EXISTENT[39m } from [32m'@env'[39m[0m
[0m [90m   | [39m                   [31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[0m
[0m [90m 2 | [39m[36mconst[39m foo [33m=[39m [32m'Bar'[39m [33m+[39m [33mBASE_URL[39m[0m

`;

exports[`@startupjs/babel-plugin-dotenv Transforms imports to constants, and places them after all other imports: Transforms imports to constants, and places them after all other imports 1`] = `

import { observer } from 'startupjs'
import { BASE_URL, STRIPE_PUBLIC_KEY } from '@env'
import React from 'react'
import { Text } from 'react-native'
const foo = 'Bar' + BASE_URL

      â†“ â†“ â†“ â†“ â†“ â†“

import { observer } from "startupjs";
import React from "react";
import { Text } from "react-native";
const STRIPE_PUBLIC_KEY = "000-000000-00";
const BASE_URL = "https://localhost:3000";
const foo = "Bar" + BASE_URL;


`;

exports[`@startupjs/babel-plugin-dotenv Transforms when @env is a sole import: Transforms when @env is a sole import 1`] = `

import { BASE_URL, STRIPE_PUBLIC_KEY } from '@env'
const foo = 'Bar' + BASE_URL

      â†“ â†“ â†“ â†“ â†“ â†“

const STRIPE_PUBLIC_KEY = "000-000000-00";
const BASE_URL = "https://localhost:3000";
const foo = "Bar" + BASE_URL;


`;

exports[`@startupjs/babel-plugin-dotenv When override is present, do it no matter if the env exists or not: When override is present, do it no matter if the env exists or not 1`] = `

import { BASE_URL as baseUrl } from '@env'
const foo = 'Bar' + baseUrl

      â†“ â†“ â†“ â†“ â†“ â†“

const baseUrl =
  (typeof window !== "undefined" &&
    window.location &&
    window.location.origin) ||
  undefined;
const foo = "Bar" + baseUrl;


`;
